### PR TITLE
merge_taxa supports dataframes

### LINKED ISSUE #210: merge_taxa: support merging dataframes
**Improvement Description**
merge_taxa should merge all information found in each file, including things like confidence scores.

**Current Behavior**
merge_taxa views the data as a series, dropping any additional feature data columns (e.g., confidence scores)

**Proposed Behavior**
view as a dataframe, so that additional columns are preserved.

### PR DESCRIPTION
**Pull request by nbokulich**
fixes #210

### CODE REVIEW

File: q2_feature_table/tests/test_merge.py [hunk 1]
```diff
@@ -286,15 +286,20 @@ def test_merge_seqs(self):
...
+    # This tests a specifically FeatureData[Taxonomy]-like dataframe
+    # and ensures delivery in valid format (Taxon column first)
 
     def test_merge_taxa(self):
         # this test calls the public API directly
-        d1 = pd.Series(['a;b;c;d', 'a;b;c;e'], index=['f1', 'f2'])
-        d2 = pd.Series(['a;b;c;d', 'a;b;c;e'], index=['f1', 'f3'])
+        d1 = pd.DataFrame([('a;b;c;d', '1.0'), ('a;b;c;e', '1.0')],
+                          index=['f1', 'f2'], columns=['Taxon', 'Confidence'])
+        d2 = pd.DataFrame([('1.0', 'a;b;c;d'), ('1.0', 'a;b;c;e')],
```
Comment (thermokarst) [inline q2_feature_table/tests/test_merge.py]: Can we make `f2` slightly different here, that way the `.combine_first` behavior is clear in the test?

File: q2_feature_table/_merge.py [hunk 2]
```diff
@@ -66,5 +66,10 @@ def merge_seqs(data: pd.Series) -> pd.Series:
     return _merge_feature_data(data)
 
 
-def merge_taxa(data: pd.Series) -> pd.Series:
-    return _merge_feature_data(data)
+def merge_taxa(data: pd.DataFrame) -> pd.DataFrame:
+    data = _merge_feature_data(data)
+    # merge orders columns alphabetically; Taxon must be first header column
+    # as defined here: https://github.com/qiime2/q2-types/blob/master/q2_types/
```
Comment (thermokarst) [inline q2_feature_table/_merge.py]: Can you make this a permalink?

File: q2_feature_table/tests/test_merge.py (hunk 1 above)
Comment (nbokulich) [reply q2_feature_table/tests/test_merge.py]: 👍

File: q2_feature_table/_merge.py (hunk 2 above)
Comment (nbokulich) [reply q2_feature_table/_merge.py]: 👍

**Comment by nbokulich** [comment]
thanks @thermokarst ! I have made those changes. ⚾️

**Comment by thermokarst** [comment]
@nbokulich, you might want to update your remotes, looks like that branch was on the canonical repo, instead of your fork.

**Pull request merged**